account_group/models/account_group.py

```diff
@@ -0,0 +1,74 @@
+# coding: utf-8
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class AccountGroup(models.Model):
+    _name = 'account.group'
+    _parent_store = True
+    _parent_order = 'name'
+    _order = 'code_prefix'
+
+    parent_id = fields.Many2one(
+        comodel_name='account.group',
+        string="Parent",
+        index=True,
+        ondelete='cascade',
+    )
+    parent_left = fields.Integer(
+        string='Left Parent',
+        index=True,
+    )
+    parent_right = fields.Integer(
+        string='Right Parent',
+        index=True,
+    )
+    name = fields.Char(
+        required=True,
+    )
+    code_prefix = fields.Char()
+    account_ids = fields.One2many(
+        comodel_name='account.account',
+        inverse_name='group_id',
+        string='Accounts',
+        help="Assigned accounts.",
+    )
+    level = fields.Integer(
+        compute='_compute_level',
+        store=True,
+    )
+
+    @api.depends('parent_id')
+    def _compute_level(self):
+        for group in self:
+            level = 1
```

Comment:
@pedrobaeza I've the feeling that this code should be simplified to avoid the recursive part and takes advantage of `store=True`..

```python
for group in self:
    group.level = 1
    parent_id = group.parent_id
    if parent_id:
        group.level = parent_id.level + 1
```

Reply:
Yeah, it can be if depends work the proper way, so I wouldn't change it without proper tests. This module will die any way in v10. Do you have a use case when this gets a performance bottleneck? My use case is 4 levels and 1200 groups, and it's very fast.

Reply:
It's just a comment. I've no use case. This code can remains as is. But as we all sometimes tend to draw inspiration from each other's code for our own developments, I think it is also important to draw attention to what can be improved to avoid potential problems. :smirk:

Reply:
Yeah, of course, and thanks for the advice. The pity is that I have already closed this development and I was very low on budget. As we have to rethink it for v11 if there's a request from a customer, I'll take into account this note for sure.